hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockManager.java

```diff
@@ -4035,17 +4039,17 @@ private void processExtraRedundancyBlock(final BlockInfo block,
     Collection<DatanodeStorageInfo> nonExcess = new ArrayList<>();
     Collection<DatanodeDescriptor> corruptNodes = corruptReplicas
         .getNodes(block);
+    boolean hasStaleStorage = false;
+    DatanodeStorageInfo staleStorage = null;
```

Comment:
this should be one set about `DatanodeStorageInfo` because it could be more than one stale storage here.